Research/lva-node-app/public/js/main.js

```diff
@@ -0,0 +1,667 @@
+* @returns {boolean} - true if able to display result in method-output-box (will be true when on output.html page)
+*/
+function displayMethodOutput(result)
+{
+  let htmlElement=document.getElementById("method-output-box");
+  if (htmlElement != null)
+  {
+    htmlElement.innerText = result;
+    return true;
+  }
+  return false;
+}
+
+/**
+* send method to invoke on LVA module
+* @async
+* @param {jsonObject} fullPayload - payload to be sent in request to server
+* @returns {Promise<any>}
+*/
+async function invokeLVAMethod(fullPayload) 
+{
+  return new Promise((resolve, reject) => 
+  {
+    let request = prepareRequest(`http://localhost:${PORT}/runMethod`);
+    // event listener. When request readyState changes, place response in the output box. If response is done (readyState=4) then update global vars accordingly
+    request.onreadystatechange = function () 
+    {
+      //on successful response. Result is object like [{methodName: 'GraphTopologyList'}, {value: 'long JSON object....'}]
+      if (request.readyState == 4 && request.status == 200) 
+      {
+        var methodName=JSON.parse(request.response)[0].method;
+        var result=JSON.stringify(JSON.parse(request.response)[1]);
+        var isFromOutputPage=displayMethodOutput(result);
+
+        if(JSON.parse(result).status < 250)
+        {
+          switch(methodName)
+          {
+            case 'GraphTopologyDelete':
+              delete graphTopologies[fullPayload.Payload.name];
+              break;
+            case 'GraphInstanceDelete':
+              delete graphInstances[fullPayload.Payload.name];
+              break;
+            case 'GraphInstanceList':
+              updateGraphsandInstances(methodName, JSON.parse(request.response)[1]);
+              //console.log("updating graphs and instances, method name: "+methodName+" and response: "+result);
+              break;
+            case 'GraphTopologyList':
+              updateGraphsandInstances(methodName, JSON.parse(request.response)[1]);
+              //console.log("updating graphs and instances, method name: "+methodName+" and response: "+result);
+              break;
+            case 'GraphInstanceSet':
+              graphInstances[fullPayload.Payload.name]=fullPayload;
+              break;
+            case 'GraphTopologySet':
+              // only alert and run display media graphs if request is coming from mediagraphs.html page
+              graphTopologies[fullPayload.Payload.name]=fullPayload;
+              if(!isFromOutputPage)
+              {
+                alert("Success!");
+                displayMediaGraphs();
+              }
+              break;
+            default:
+              break;
+          }
+          resolve(result);
+        }
+        else
+        {
+          reject(result);
+        }
+      }
+      else if (request.readyState == 4)
+      {
+        alert(request.response);
+        reject(request.response);
+      }
+      else
```

Comment:
what is this case? i don't see an explicit reject of the promise but it seems wrong. why would we just display? seems like this should be merged with the state above?

Reply:
merged with state above. Pushing now